components/organisms/ContinueRegistrationModalContent.tsx

```diff
@@ -0,0 +1,152 @@
+import { RegistrationStep } from "@/lib/name-registration/constants";
+import { LocalRegistrationNameData } from "@/lib/name-registration/types";
+import { useNameRegistration } from "@/lib/name-registration/useNameRegistration";
+import { BasicInfoKey, SocialAccountsKeys } from "../02-molecules";
+import { endNameRegistrationPreviouslyOpen } from "@/lib/name-registration/localStorage";
+import { useAccount } from "wagmi";
+import { Dispatch, SetStateAction } from "react";
+import { Button } from "@ensdomains/thorin";
+
+export const ContinueRegistrationModalContent = ({
+  name,
+  localNameRegistrationData,
+  setIsModalContinueRegistrationOpen,
+}: {
+  name: string;
+  localNameRegistrationData: LocalRegistrationNameData;
+  setIsModalContinueRegistrationOpen: Dispatch<SetStateAction<boolean>>;
+}) => {
+  const {
+    setRegistrationYears,
+    setCurrentRegistrationStep,
+    setAsPrimaryName,
+    setEnsResolver,
+    setTextRecords,
+    setCommitSubmitTimestamp,
+    setDomainAddresses,
+    nameRegistrationData,
+  } = useNameRegistration();
+
+  const { address } = useAccount();
+
+  const continueRegistration = () => {
+    //Set State of Registration Years
+    if (
+      localNameRegistrationData.registrationYears !== undefined &&
+      localNameRegistrationData.registrationYears >= 1
+    ) {
+      setRegistrationYears(localNameRegistrationData.registrationYears);
+    } else {
+      setCurrentRegistrationStep(RegistrationStep.RegistrationYears);
+      return;
+    }
```

Comment:
It seems this else blocks are not needed since these are setting the default state of these fields, which should already be there

Reply:
These else blocks are setting the Step to the correct step according to the localNameRegistrationData stored values;